sw/device/silicon_creator/lib/manifest_def.c

```diff
@@ -9,14 +9,25 @@
 /*
  * Declarations for the manifest fields populdated by the linker script.
  */
-extern const uint32_t _manifest_code_start[];
-extern const uint32_t _manifest_code_end[];
-extern const uint32_t _manifest_entry_point[];
+extern char _manifest_code_start[];
+extern char _manifest_code_end[];
+extern char _manifest_entry_point[];
 
-static_assert(sizeof(uint32_t) == sizeof(uintptr_t), "Pointer is not 32-bits.");
+static_assert(sizeof(uint32_t) == sizeof(char *), "Pointer is not 32-bits.");
```

Comment:
Was this assertion actually needed in the first place? The externs above don't contain pointers to the data, they directly refer to the data, but it seems like this assertion was written assuming otherwise, and trying to protect against the case the pointer size changed. But there never was a pointer involved in the first place (besides the address/pointer that necessarily exists when you materialize the address of a global).

Reply:
I agree, these are basically offsets calculated in the linker script. You're right, I think it makes sense to remove this assertion.